lib/logger.js

```diff
@@ -81,9 +81,20 @@ function createLogger (options) {
     logger.child = () => logger
     return { logger, hasLogger: false }
   } else {
-    options.logger = typeof options.logger === 'object' ? options.logger : {}
-    options.logger.level = options.logger.level || 'info'
-    options.logger.serializers = Object.assign({}, serializers, options.logger.serializers)
+    const ops = {}
+    if (typeof options.logger === 'object') {
```

Comment:
```suggestion
    if (Object.prototype.toString.call(options.logger) === '[object Object]') {
```

Reply:
Hi, thanks you suggestion😁，while I think It is more concise to use typeof whether it is an object, I want know using below benefit ？ 
```
Object.prototype.toString.call(options.logger) === '[object Object]'
```

Reply:
`typeof [] === 'object'` is `true`

Reply:
`typeof null === 'object'` is `true`

Reply:
thanks